econml/grf/_base_grf.py

```diff
@@ -793,10 +793,12 @@ def predict_full(self, X, interval=False, alpha=0.05):
         """
         if interval:
             point, pred_var = self._predict_point_and_var(X, full=True, point=True, var=True)
+            pred_var = np.maximum(pred_var, 1e-32)

```

Comment:
If the theory is that this is only necessary because of near-zero negative values, it might be better to also assert that explicitly first (e.g. something like `assert np.all(np.isclose(pred_var[pred_var<0], 0, atol=1e-16))`); any failed assertion would be an indication that our `_predict_point_and_var` is doing something wonky (rather than anything the user did wrong), but if we're getting big negative values then it doesn't seem obvious that just setting them to a small positive value is the right thing to do.

Reply:
Sounds fair, I'll send a commit once done

Reply:
The test case that causes this problem uses real data so I'll have to make a synthetic dataset that can be used to create the same problem. 

Unless you're suggesting I add that assert inline with the code?

Reply:
@kbattocchi ^^

Reply:
@arose13 I'm so sorry, I saw your first comment but didn't notice the second one. Yes, my suggestion is to add the assert just before line 796.  If it's infeasible to quickly come up with a test where your change affects the outcome then don't worry about it - it would be nice to have but isn't critical.

Reply:
No stress at all. The assert statement is easy to do. I'm not sure if I can make a good test case for this however. If you're fine with 2 PRs I can submit it that way

Reply:
Added the assertion. I'vve never be able to produce large negative values in my tests so I'm hoping that assertion is never triggered in real life.